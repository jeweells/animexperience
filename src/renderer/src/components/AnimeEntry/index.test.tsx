import { AnimeEntry, AnimeEntryProps } from './index'

describe('AnimeEntry', () => {
  const returnTestId = 'render-return'
  const props = {
    onClick: jest.fn(),
    onMouseOut: jest.fn(),
    onMouseOver: jest.fn(),
    render: jest.fn(() => <div data-testid={returnTestId} />),
    isPopover: false,
    index: 0,
    visible: false,
    sliding: false
  }
  beforeEach(() => {
    props.onClick.mockClear()
    props.onMouseOut.mockClear()
    props.onMouseOver.mockClear()
    props.render.mockClear()
  })

  const getComponent = (override: Partial<AnimeEntryProps> = {}) => (
    <AnimeEntry {...props} {...override} />
  )

  it('renders default', async () => {
    const wrapper = render(getComponent())
    await waitFor(() => expect(wrapper.getByTestId(returnTestId)).toBeInTheDocument())
    expect(props.render).toHaveBeenCalled()
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it.each([1, 2])('should take longer to appear as long as index (%s) increases', async (index) => {
    const wrapper = render(getComponent({ index }))
    await waitFor(() => expect(wrapper.getByTestId(returnTestId)).toBeInTheDocument())
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('renders when its a popover', async () => {
    const wrapper = render(getComponent({ isPopover: true }))
    await waitFor(() => expect(wrapper.getByTestId(returnTestId)).toBeInTheDocument())
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('renders when hovered', async () => {
    // TODO: Implement, popover should be visible
  })

  it('renders when hovered but sliding', async () => {
    // TODO: Implement, popover should be hidden
  })

  it('can click when hovered', async () => {
    // TODO: Implement, popover should be clicked
  })
})
