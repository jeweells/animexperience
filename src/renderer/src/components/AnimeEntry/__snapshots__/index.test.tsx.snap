// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimeEntry renders default 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  border-radius: 4px;
  overflow: hidden;
  --width: calc( ( 100vw - 3 * 16px - 2 * 72px ) / 4 );
  --height: calc(var(--width) * 0.64);
  width: var(--width);
  height: var(--height);
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  cursor: pointer;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover {
  z-index: 10;
}

@media (max-width:1000px) {
  .c0 {
    --width: calc( ( 100vw - 2 * 16px - 2 * 72px ) / 3 );
    --height: calc(var(--width) * 0.64);
  }
}

@media (max-width:800px) {
  .c0 {
    --width: calc( ( 100vw - 1 * 16px - 2 * 72px ) / 2 );
    --height: calc(var(--width) * 0.64);
    --height: calc(var(--width) * 0.84);
  }
}

@media (max-width:500px) {
  .c0 {
    --width: calc( ( 100vw - 0 * 16px - 2 * 72px ) / 1 );
    --height: calc(var(--width) * 0.64);
  }
}

<div
    data-testid="anime-entry-wrapper"
    style="opacity: 1; webkit-transition: opacity 1000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 1000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <div
      class="c0"
      data-testid="anime-entry-background"
    >
      <div
        data-testid="render-return"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`AnimeEntry renders when its a popover 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  border-radius: 4px;
  overflow: hidden;
  --width: calc( ( 100vw - 3 * 16px - 2 * 72px ) / 4 );
  --height: calc(var(--width) * 0.64);
  width: var(--width);
  height: var(--height);
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  cursor: pointer;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover {
  z-index: 10;
}

@media (max-width:1000px) {
  .c0 {
    --width: calc( ( 100vw - 2 * 16px - 2 * 72px ) / 3 );
    --height: calc(var(--width) * 0.64);
  }
}

@media (max-width:800px) {
  .c0 {
    --width: calc( ( 100vw - 1 * 16px - 2 * 72px ) / 2 );
    --height: calc(var(--width) * 0.64);
    --height: calc(var(--width) * 0.84);
  }
}

@media (max-width:500px) {
  .c0 {
    --width: calc( ( 100vw - 0 * 16px - 2 * 72px ) / 1 );
    --height: calc(var(--width) * 0.64);
  }
}

<div
    data-testid="anime-entry-wrapper"
    style="opacity: 1;"
  >
    <div
      class="c0"
      data-testid="anime-entry-background"
    >
      <div
        data-testid="render-return"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`AnimeEntry should take longer to appear as long as index (1) increases 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  border-radius: 4px;
  overflow: hidden;
  --width: calc( ( 100vw - 3 * 16px - 2 * 72px ) / 4 );
  --height: calc(var(--width) * 0.64);
  width: var(--width);
  height: var(--height);
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  cursor: pointer;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover {
  z-index: 10;
}

@media (max-width:1000px) {
  .c0 {
    --width: calc( ( 100vw - 2 * 16px - 2 * 72px ) / 3 );
    --height: calc(var(--width) * 0.64);
  }
}

@media (max-width:800px) {
  .c0 {
    --width: calc( ( 100vw - 1 * 16px - 2 * 72px ) / 2 );
    --height: calc(var(--width) * 0.64);
    --height: calc(var(--width) * 0.84);
  }
}

@media (max-width:500px) {
  .c0 {
    --width: calc( ( 100vw - 0 * 16px - 2 * 72px ) / 1 );
    --height: calc(var(--width) * 0.64);
  }
}

<div
    data-testid="anime-entry-wrapper"
    style="opacity: 1; webkit-transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <div
      class="c0"
      data-testid="anime-entry-background"
    >
      <div
        data-testid="render-return"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`AnimeEntry should take longer to appear as long as index (2) increases 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  border-radius: 4px;
  overflow: hidden;
  --width: calc( ( 100vw - 3 * 16px - 2 * 72px ) / 4 );
  --height: calc(var(--width) * 0.64);
  width: var(--width);
  height: var(--height);
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  cursor: pointer;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover {
  z-index: 10;
}

@media (max-width:1000px) {
  .c0 {
    --width: calc( ( 100vw - 2 * 16px - 2 * 72px ) / 3 );
    --height: calc(var(--width) * 0.64);
  }
}

@media (max-width:800px) {
  .c0 {
    --width: calc( ( 100vw - 1 * 16px - 2 * 72px ) / 2 );
    --height: calc(var(--width) * 0.64);
    --height: calc(var(--width) * 0.84);
  }
}

@media (max-width:500px) {
  .c0 {
    --width: calc( ( 100vw - 0 * 16px - 2 * 72px ) / 1 );
    --height: calc(var(--width) * 0.64);
  }
}

<div
    data-testid="anime-entry-wrapper"
    style="opacity: 1; webkit-transition: opacity 2000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 2000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <div
      class="c0"
      data-testid="anime-entry-background"
    >
      <div
        data-testid="render-return"
      />
    </div>
  </div>
</DocumentFragment>
`;
